pkg/controller/garbagecollector/patch.go

```diff
@@ -32,16 +32,20 @@ import (
 // getMetadata tries getting object metadata from local cache, and sends GET request to apiserver when
 // local cache is not available or not latest.
 func (gc *GarbageCollector) getMetadata(apiVersion, kind, namespace, name string) (metav1.Object, error) {
```

Comment:
add a unit test that exercises the following error cases:

1. attempt to get a namespaced object without providing a namespace (should return namespacedOwnerOfClusterScopedObjectErr where it would previously attempt the API request and get a 404)
2. attempt to get a cluster-scoped type passing in a namespace (should ignore the passed-in namespace and fetch the object successfully where it would previously attempt the fetch namespaced and 404)